lib/data/datasource_impl/contact/phonebook_contact_datasource_impl.dart

```diff
@@ -25,5 +26,31 @@ class PhonebookContactDatasourceImpl implements PhonebookContactDatasource {
           ],
         )
         .toList();
+    
+    return removeDuplicatedPhoneNumbers(listAllContacts);
+  }
+
+  List<Contact> removeDuplicatedPhoneNumbers(List<Contact> listContacts) {
+    final listVisitedPhoneNumbers = <String>[];
+    final listFilteredContacts = <Contact>[];
+
+    for (final contact in listContacts) {
+      final phoneNumber = contact.phoneNumber;
+      if (phoneNumber != null) {
+        final normalizedPhoneNumber = normalizePhoneNumber(phoneNumber);
+        if (listVisitedPhoneNumbers.contains(normalizedPhoneNumber)) {
+          continue;
+        } else {
+          listVisitedPhoneNumbers.add(normalizedPhoneNumber);
+          listFilteredContacts.add(contact);
+        }
+      }
+    }
+
+    return listFilteredContacts;
+  }
+
+  String normalizePhoneNumber(String phoneNumber) {
```

Comment:
this function should be in other `utils` file, to make it reusable.

Reply:
dont forget to write test for this funciton

Reply:
done